synapse/storage/data_stores/main/devices.py

```diff
@@ -158,6 +187,16 @@ def get_devices_by_remote(self, destination, from_stream_id, limit):
                 # Stop processing updates
                 break
 
+            # skip over cross-signing keys
+            if (
+                update[0] in master_key_by_user
```

Comment:
I suggest unpacking the `update` into useful variable names rather than repeatedly indexing into it.